csharp/extractor/Semmle.Extraction.CSharp/Entities/Expressions/Invocation.cs

```diff
@@ -33,7 +37,7 @@ protected override void PopulateExpression(TextWriter trapFile)
             var target = TargetSymbol;
             switch (Syntax.Expression)
             {
-                case MemberAccessExpressionSyntax memberAccess:
+                case MemberAccessExpressionSyntax memberAccess when Kind == ExprKind.METHOD_INVOCATION || IsEventDelegateCall() || IsExplicitDelegateInvokeCall():
```

Comment:
When is this `when` condition not the same as `Kind == ExprKind.METHOD_INVOCATION || Kind = ExprKind.DELEGATE_INVOCATION`?

Reply:
Exactly when the member being accessed is not an event or a direct `.Invoke`, i.e., when it is a field or a property.